griptape/common/decorators.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from typing import Any, Callable
+
+import wrapt
+
+
+def observable(*dargs: Any, **dkwargs: Any) -> Any:
+    @wrapt.decorator
+    def decorator(wrapped: Callable, instance: Any, args: Any, kwargs: Any) -> Any:
```

Comment:
Totally forgot that all that complexity was due to avoiding wrapt, love it

As for unit testing, how about diffing the attributes w and wo wrapt to find what was needed to solve the issues